test/Parser.js

```diff
@@ -528,6 +528,78 @@ describe('Parser', () => {
     ).to.equal(`.user h1:matches(.a,.b,:not(.c)){display:none;}`)
   })
 
+  test('pseudo-selectors', () => {
+    expect(
+      stylis(`
+        :not(&) {
+          color: red;
+        }
+        :matches(&) {
+          color: red;
+        }
+        :is(&) {
+          color: red;
+        }
+        :any(&) {
+          color: red;
+        }
+        :where(&) {
+          color: red;
+        }
+        :has(&) {
+          color: red;
+        }
+      `)
+    ).to.equal([
+      `:not(.user){color:red;}`,
+      `:matches(.user){color:red;}`,
+      `:is(.user){color:red;}`,
+      `:any(.user){color:red;}`,
+      `:where(.user){color:red;}`,
+      `:has(.user){color:red;}`
+    ].join(''))
+  })
+
+  test('nested pseudo-selectors', () => {
+    expect(
+      stylis(`
```

Comment:
I think it's the correct output here - the interesting bit is that Sass doesn't seem to handle this correctly

Reply:
I think this is because `:is` and `:where` are broken:

```css
.foo {
  :not(&) {
    color: red;
  }

  :matches(&) {
    color: red;
  }

  :is(&) {
    color: red;
  }

  :any(&) {
    color: red;
  }

  :where(&) {
    color: red;
  }

  :has(&) {
    color: red;
  }
}
```

```css
:not(.foo) {
  color: red;
}
:matches(.foo) {
  color: red;
}
.foo :is(&) {
  color: red;
}
:any(.foo) {
  color: red;
}
.foo :where(&) {
  color: red;
}
:has(.foo) {
  color: red;
}
```

If I remove those, it looks like works:

```css
.foo {
    :not(&) {
        :matches(&) {
            :any(&) {
                :has(&) {
                    color: red;
                }
            }
        }
    }
}
```

```css
:has(:any(:matches(:not(.foo)))) {
  color: red;
}
```

(using https://www.sassmeister.com/)